consul/src/main/scala/io/buoyant/consul/CatalogNamer.scala

```diff
@@ -2,16 +2,30 @@ package io.buoyant.consul
 
 import com.twitter.finagle._
 import com.twitter.util._
-import io.buoyant.consul.v1.{UnexpectedResponse, ServiceNode}
+import io.buoyant.consul.v1.{ServiceNode, UnexpectedResponse}
+import io.buoyant.namer.Metadata
 
 class CatalogNamer(
   idPrefix: Path,
-  api: v1.CatalogApi,
-  includeTag: Boolean = false
+  catalogApi: v1.CatalogApi,
+  agentApi: v1.AgentApi,
+  includeTag: Boolean = false,
+  setHost: Boolean = false
 ) extends Namer {
 
   import CatalogNamer._
 
+  private[this] lazy val domainFuture: Future[String] =
```

Comment:
Hm, sounds more complicated than I though first. In this case, would you mind making it 'just an another Future call each time' for now. Once we will get some usage experience with this we will be able to see the performance impact and if it will be significant then we will contribute another PR with an improvement.

What do you think?